### ISSUE TITLE
The default extension of ".json" seems more annoying than useful

### ISSUE THREAD

**Issue by tobyink** [srchulo/JSON-Slurper#2]
What if I actually want to read a file called "ref", not "ref.json"? It seems I simply cannot do this with JSON::Slurper.

**Comment by srchulo** [comment]
Yes, sorry. I added the current default because it was useful to me, and I imagined a lot of others, with really the only case where it doesn't work is if you don't have any extension. Do you need this yourself, or is this just feedback? I can add the ability to just access `ref`, and if the simplest thing is to just require the full filename, I'm comfortable making a breaking change since I just released the module this week.

**Comment by tobyink** [comment]
Making common use cases easier is a good thing, but the current implementation makes reading a file called "ref" with JSON::Slurper not just hard, but impossible. And it means that if I've got a variable called `$f` that I know contains an exact file name, without me inspecting `$f` first, I can't know whether I can safely pass it to JSON::Slurper or not.

This is mostly just general feedback. I don't have any immediate plans to use JSON::Slurper; I just like to check out potentially useful new modules occasionally. For now, I'll probably be sticking with `decode_json(path($f)->slurp_utf8)` using JSON::MaybeXS and Path::Tiny.

**Comment by srchulo** [comment]
Yeah, that's good feedback. I think the case where you would have to inspect `$f` that your passed is a compelling reason to change the default. Maybe I'll make what most people would expect (just using the filename passed) the default, and add a flag for if you want to have the extension automatically added, something like:

```
use JSON::Slurper qw(-auto_ext slurp_json spurt_json);
```

And add it as an option for the object interface:

```
JSON::Slurper->new(auto_ext => 1);
```

**Comment by srchulo** [comment]
@tobyink thank you for the feedback! The default now is to not automatically add `.json`. `.json` is only added when you specify `auto_ext` as mentioned in the comment above.

Thanks again. I think this makes more sense :)

**Comment by tobyink** [comment]
You might want to consider using `%AUTO_EXT` and `$AUTO_EXT{+caller()}` instead of `$AUTO_EXT`.

**Comment by srchulo** [comment]
@tobyink ah, that's a good suggestion! Is the `+` necessary if you're going to put the parentheses on `caller()`?

**Comment by srchulo** [comment]
Thanks again! #3 was fixed here 5e2f387
